lib/stylus-globals.js

```diff
@@ -54,17 +55,193 @@ function getCurlStatement (baseUri, method, resource) {
   const parentUrl = resource.parentUrl || ''
   const relativeUri = resource.relativeUri || '/'
 
-  const headers = method.headers
+  let headers = method.headers
     .filter((header) => header.examples && header.examples.length !== 0)
-    .map((header) => ` \\\n\t-H "${header.key}: ${header.examples[0].value}"`).join('')
+    .map((header) => ` \\\n\t-H "${header.key}: ${header.examples[0].value}"`)
 
   let params = method.queryParameters
     .filter((param) => param.examples && param.examples.length !== 0)
     .map((param) => `${param.key}=${param.examples[0].value}`)
-    .join('&')
-  params = params === '' ? '' : `?${params}`
 
-  return `curl -X ${method.method.toUpperCase()} "${baseUri}${parentUrl}${relativeUri}${params}"${headers}${payload}`
+  const resourceOptions = {
+    method: method.method,
+    baseUri: baseUri,
+    path: `${parentUrl}${relativeUri}`,
+    params: params,
+    headers: headers,
+    payload: payload
+  }
+
+  const commands = []
+  if (securitySchemes && method.securedBy) {
+    method.securedBy.forEach((methodScheme) => {
+      if (methodScheme) {
```

Comment:
this could be
```javascript
if (methodScheme && securitySchemes[methodScheme.schemeName])
```
That way you get rid of one of the else blocks (both elses seem to do the same)

Reply:
Hmmm. One of the `else`s is nested inside a `forEach` loop, which loops over all the security schemes for the method. That inner else guards against the scenario where the method has named a security scheme, but a matching security scheme hasn't been defined at the root. Of course, this would be an invalid RAML document, but I figured we should guard against it.

Reply:
It also handles the case where multiple security schemes are defined on the method, and one of them is explicitly `null`. The test case for this is on line 433 of stylus-globals.test.js.